summaries/search-based-survey.md

```diff
@@ -0,0 +1,335 @@
+> Large search space ---constraints---> Reduction of search space
+
+#### Dynamic Domain Reduction
+Dynamic Domain Reduction is a *static analysis technique* that starts with domains of the input variables and reduce these dynamically during the symbolic execution stage in contrast to normal Domain Reduction. 
+
+*This technique still suffers from difficulties with computed storage locations, loops and non-ordinal variable types(e.g. Enums).*
+
+### Dynamic Structural Test Data Generation
+*Dynamic methods* use some input to execute a program and uses instrumentation to observer the results.
+Many of the issues regarding Symbolic Execution are resolved by directly resolving e.g. pointer values and array subscripts.
+
+#### Random Testing
+Executes the program with random inputs and observes executed program structures.
+When using this technique, code that has low probability of being executed is often not covered, in such cases more directed search techniques are required.
+
+##### Applying Local Search
+In the Local Search technique, the tester selects a path through the program and produces a straight-line version of that path.
+Path constraints are constructed with constant for each constraint estimating the satisfiability of the constraint.
+A function *f* is constructed out of all of these constraints.
+*f* provides an estimate of the satisfiability of all constraints.
+Using numerical maximisation techniques, the input values are maximized to satisfy as many constraints as possible.
+One major draw-back of this was that that run-time errors could occur, since the dependency between constraints were not taken into account, which could create sequences, which in practice are impossible to execute.
+
+Improvements to this technique emerged to resolve this problem.
+The search was now targeted with the satisfaction of each constraint instead of all the constraints. 
+If during the execution an undesired branch is take, a local search is started, that is targeted to minimize branch distance of the desired branch, also known as the *branch distance*.
+Creating these inputs is done using the *alternating variable* method, that starts with an *exploratory phase* in which probes neighbouring constraints by increase or decreasing only that specific variable.
+If the move leads to an improved objective value, a *pattern phase* is entered.
+In this phase a larger move is made in the direction of the improvement, this is repeated until a minimum for that objective function is found for that variable.
+After this is complete, other variable are explored.
+
+These techniques suffer however from one main problem: the final solution is highly dependent on the starting solution, which can be very wasteful if unfavorable values are chosen.
+A solution to this is the creation of an *influences graph*, that is created via dynamic data flow analysis and is used to detect which input variable influence the outcome at a branching node.
```

Comment:
I'm blaming the author on this one.
> Korel’s work makes use of extra information derived from the program, in the form of an **“influences”** graph. An **influences** graph is used to detect which input variables are able to influence the outcome at the current branching node, as determined using dynamic data flow analysis.

Could of course change it if you want to though.